### PR TITLE
Softmax operation implementation for mkldnn 

### PR DESCRIPTION
**Pull request by olegsemeniv**
**Softmax operation implementation on mkldnn**

Added softmax operation support

Tested:
- manually
- some test cases added

### CODE REVIEW

File: libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+                const int xRank = x->rankOf();
+                bool bSupportedRanks = (xRank > 2 && xRank < 7);
+                /*
+                Source     Destination
+                f32 	    f32
+                f16 	    f16
+                */
+                return  block.isUseMKLDNN() && bSupportedRanks && ((xType == DataType::FLOAT32 && zType == DataType::FLOAT32)
+                    || (xType == DataType::BFLOAT16 && zType == DataType::BFLOAT16));
```
Comment (raver119) [inline libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp]: According to the documentation, FLOAT32 and FLOAT16 only supported. BFLOAT16 isn't mentioned.

File: libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                }
+                else if (5 == xRank) {
+                    format = dnnl::memory::format_tag::abcde;
+                }
+                else if (6 == xRank) {
+                    format = dnnl::memory::format_tag::abcdef;
+                }
+
+                dnnl::memory::data_type xType = (x->dataType() == DataType::FLOAT32) ? dnnl::memory::data_type::f32 : dnnl::memory::data_type::bf16;
+                dnnl::memory::data_type zType = (z->dataType() == DataType::FLOAT32) ? dnnl::memory::data_type::f32 : dnnl::memory::data_type::bf16;
```
Comment (raver119) [inline libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp]: bf16 isn't supported according to MKLDNN docs. Please change to f16

**Review by raver119 — CHANGES_REQUESTED** [review]
LGTM, minor fix required

File: libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp (hunk 1 above)
Comment (olegsemeniv) [reply libnd4j/include/ops/declarable/platform/mkldnn/softmax.cpp]: as discussed via slack lived only FLOAT32

**Review by raver119 — APPROVED** [review]

**Pull request merged**